### PR TITLE
add retrofit defintions

### PR DESCRIPTION
**Pull request by longshuicy**
This PR added Fragility Mapping Entry keys where we can utilize to properly define fragilities and retrofit. 

To test this PR:
1. POST to  /dfr3/api/mappings with below body as `application/json`.
Use the below json for test body
[galveston_sw_plus_retrofit_mapping.json](https://github.com/IN-CORE/incore-services/files/14316401/galveston_sw_plus_retrofit_mapping.json)
[galveston_flood_plus_retrofit_mapping.json](https://github.com/IN-CORE/incore-services/files/14316402/galveston_flood_plus_retrofit_mapping.json)
[SLC_plus_retrofit.json](https://github.com/IN-CORE/incore-services/files/14316405/SLC_plus_retrofit.json)

2. GET /dfr3/api/mappings and try GET /mappings/{ID posted from step 1)

3. DELETE /dfr3/api/mappings/{ID posted from step 1}

### CODE REVIEW

File: server/dfr3-service/src/main/java/edu/illinois/ncsa/incore/service/dfr3/models/MappingEntryKey.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2023 University of Illinois and others.  All rights reserved.
```
Comment (navarroc) [inline server/dfr3-service/src/main/java/edu/illinois/ncsa/incore/service/dfr3/models/MappingEntryKey.java:2]: Can you fix the year?

File: server/dfr3-service/src/main/java/edu/illinois/ncsa/incore/service/dfr3/models/MappingEntryKeyConfig.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package edu.illinois.ncsa.incore.service.dfr3.models;
```
Comment (navarroc) [inline server/dfr3-service/src/main/java/edu/illinois/ncsa/incore/service/dfr3/models/MappingEntryKeyConfig.java:1]: Should add the license header

File: server/dfr3-service/src/main/java/edu/illinois/ncsa/incore/service/dfr3/models/MappingEntryKey.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+package edu.illinois.ncsa.incore.service.dfr3.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import dev.morphia.annotations.Embedded;
+
+@Embedded
+public class MappingEntryKey {
+    private String name;
+    private String description;
+    private Boolean defaultKey;
```
[the remark this answers is not in the document]
Comment (longshuicy) [reply server/dfr3-service/src/main/java/edu/illinois/ncsa/incore/service/dfr3/models/MappingEntryKey.java:20]: oh i realize "default" is a Java preserved name so i can't use "default". I will update the design.

File: server/dfr3-service/src/main/java/edu/illinois/ncsa/incore/service/dfr3/models/MappingEntryKeyConfig.java (hunk 2 above)
Comment (longshuicy) [reply server/dfr3-service/src/main/java/edu/illinois/ncsa/incore/service/dfr3/models/MappingEntryKeyConfig.java:11]: Done

**Review by ylyangtw — APPROVED** [review]
Tests pass. Changes look good.

**Review by Rashmil-1999 — APPROVED** [review]
Looks good!

**Pull request merged**